### PR TITLE
LidarLite port to raspberrypi

### PR DESCRIPTION
**Pull request by dagar**
@belaidH I squashed your changes onto latest master and fixed a few things. The main thing you were missing is opening the PX4 virtual device (px4_open, px4_close, px4_ioctl, etc) instead of the real version when appropriate. It's rather unintuitive.

Can you give this another try?

### CODE REVIEW

**Comment by belaidH** [comment]
Thank you very much. I tried it and this is the result

2018-01-09 2:50 GMT+01:00 Daniel Agar <notifications@github.com>:

**Comment by belaidH** [comment]
I have downloaded the patch with this command:
curl -L
https://patch-diff.githubusercontent.com/raw/PX4/Firmware/pull/8633.patch >
/tmp/patch/new.patch

and applied it with:
git apply  /tmp/patch/new.patch



2018-01-09 8:21 GMT+01:00 Hamoum Belaid <belaidhamoum@gmail.com>:

**Comment by belaidH** [comment]
and this print is infinite

2018-01-09 8:23 GMT+01:00 Hamoum Belaid <belaidhamoum@gmail.com>:

**Comment by belaidH** [comment]
running perfectly 
distance_sensor

File: src/drivers/ll40ls/LidarLitePWM.h [hunk 1]
```diff
@@ -32,7 +32,7 @@
  ****************************************************************************/
 
 
-/**
+/*
```
Comment (bkueng) [inline src/drivers/ll40ls/LidarLitePWM.h]: Nitpick: this was correct before (doxygen comment)

File: src/drivers/ll40ls/ll40ls.cpp [hunk 2]
```diff
@@ -135,37 +135,37 @@ void start(enum LL40LS_BUS busid, uint8_t rotation)
...
+		PX4_WARN("No LidarLite found");
 		return;
 	}
 
-	fd = open(instance->get_dev_name(), O_RDONLY);
+	fd = px4_open(instance->get_dev_name(), O_RDONLY);
 
 	if (fd == -1) {
-		warnx("Error opening fd");
+		PX4_WARN("Error opening fd");
```
Comment (bkueng) [inline src/drivers/ll40ls/ll40ls.cpp]: Why a warning and not an error (others too)?

Comment (dagar) [reply src/drivers/ll40ls/ll40ls.cpp]: No reason in particular, these were done in separate passes by different people. I'll give it another look.

Comment (dagar) [reply src/drivers/ll40ls/ll40ls.cpp]: Fixed.

**Pull request merged**